backend/src/main/java/org/cryptomator/hub/api/VaultResource.java

```diff
@@ -374,6 +373,7 @@ public Response grantAccess(@PathParam("vaultId") UUID vaultId, @PathParam("user
 	@GET
 	@Path("/{vaultId}")
 	@RolesAllowed("user")
+	// @VaultRole(VaultAccess.Role.MEMBER) // TODO: members and admin may do this...
```

Comment:
No, hence the TODO 😉

I think we should use separate methods for admin and members.

In fact I believe for the admin interface the frontend really just makes use of `getAll`...?

Reply:
This method is used in the frontend when retrieving data for the "vault details" view. An admin is able to list all vaults, regardless of membership. This means that this method is indeed called by admins _and_ members.

In this case, I think separate methods would be better. But that would require some if/else logic in the frontend, though. Maybe it's better this way because admins (who are not members) might need a different view for things like "claim ownership via recovery key" anyway (if the vault somehow has no owner anymore).

Reply:
Maybe we need some annotation like this, but this exceeds the scope of this PR:

```java
@AuthRequireEither({
    @RolesAllowed("admin"),
    @AuthRequireAll({
        @RolesAllowed("user"),
        @VaultRole(VaultAccess.Role.MEMBER)
    })
})
```

Reply:
All right, let's keep this TODO then and don't blow up this PR.